phases/10-llms-from-scratch/09-constitutional-ai-self-improvement/code/main.py

```
@@ -0,0 +1,251 @@
+                "problems": crit["problems"],
+                "revised": revised,
+                "changed": revised != response,
+            }
+        )
+    return revised_pairs
+
+
+def reward_math(prompt: str, response: str) -> float:
+    try:
+        cleaned = prompt.replace("What is ", "").replace("?", "").strip()
+        expected = eval(cleaned, {"__builtins__": {}}, {})
```

Ruff (0.15.10)

[error] Use of possibly insecure function; consider using `ast.literal_eval`

(S307)